core/src/robot/Robot.cc

```diff
@@ -1323,6 +1323,10 @@ namespace jiminy
                                Eigen::Ref<vectorN_t const> const & a,
                                vectorN_t                   const & u)
     {
+        // Update kinematic quantities before updating sensors.
+        // There is no need to update frame placement as this has already been done
+        // before.
+        pinocchio::forwardKinematics(pncModel_, pncData_, q, v, a);
```

Comment:
I didn't spend too much time thinking this though and simply added a forwardKinematics step here.

Indeed, even though for an unconstrain simulation we may hope to mutalize with previous computation, when using constraints we call last forwardKinematics(q, v, 0). So we need to set the state back to (q, v, a) at one point: that way, we are sure things are fine when computing the sensor state.

Reply:
It's fine. Too much optimization kills maintainability and flexibility. But comparing with `computeForwardKinematics`, why are you not calling `updateFramePlacements` ?

Reply:
>  why are you not calling `updateFramePlacements`

It's not necessary, but you are right, adding it for clarity. This function doesn't cost much anyway (contrary to framesForwardKinematics which I had in mind, update does not perform any kinematic computation).